test/custom_task_test.go

```diff
@@ -729,9 +729,10 @@ func resetConfigMap(ctx context.Context, t *testing.T, c *clients, namespace, co
 func getFeatureFlagsBaseOnAPIFlag(t *testing.T) *config.FeatureFlags {
 	t.Helper()
 	alphaFeatureFlags, err := config.NewFeatureFlagsFromMap(map[string]string{
-		"enable-api-fields":         "alpha",
-		"results-from":              "sidecar-logs",
-		"enable-tekton-oci-bundles": "true",
+		"enable-api-fields":            "alpha",
+		"results-from":                 "sidecar-logs",
+		"enable-tekton-oci-bundles":    "true",
+		"enable-cel-in-whenexpression": "true",
```

Comment:
Why is this in a custom task test? I thought it might be something we need for pipeline tests.

Reply:
Yeah I totally agree! This helper function is not clear to developers and it is the reason I failed the e2e tests. I can clean this up in another PR. Will create an issue to track

Reply:
Sounds good! Thanks

Reply:
Just for the record, this is the issue tracking this cleanup: https://github.com/tektoncd/pipeline/issues/7269